### PR TITLE
Blog about performance update

### PR DESCRIPTION
**Pull request by juliakreger**

### CODE REVIEW

File: src/pages/blog/performance-update.md [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+Not only does the scale differ, but so do the usage patterns, and ultimately the resources required to continue to manage a dynamic Bare Metal cloud. In this exercise, this was something we came to learn with our code.
+
+We treated database queries as one size fits all, and to put it bluntl, they cannot be treated as such. Without witing more on the subject than anyone would want to read, there are trade-offs. Ironic now attempts to leverage the trade-off when lists of nodes are being requested which *should* improve general node list performance. The trade off being in those cases, Ironic will no longer use a database join to pull in a composite field for a node. This is because that join was actually triggering additional overhead in the SQL client as the result set was being downloaded and converted for use by the program. Basic performance testing shows this change to be on-par with when we utilzied joins, but it should help operators who use one or more node traits assigned to a node.
```
Comment (jayofdoom) [inline src/pages/blog/performance-update.md:44]: spelling:
 - bluntl -> blunt
 - witing -> writing
 -

File: src/pages/blog/performance-update.md [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+<br>
+* Make **chassis_uuid** use a join - Presently, explicitly asking for a node's chassis_uuid is kind of slow. Unfortunately this is an object model change and cannot be backported.
+* Make **allocation_uuid** use a join - Explicitly asking for an allocation, also causes some additional overhead query. Unfortunately, this is also an object model change and cannot be backported.
+* Improve Nova's interaction with nova.virt.ironic - This is not explicitly in Ironic, but the driver for Ironic in nova is subjected to some calls internally that make sense when your nova-compute service is a hypervisor, and not a transport through to Ironic. Specifically [bug 1923955](https://bugs.launchpad.net/nova/+bug/1933955) is one of those cases. We believe this can use the internal cache.
+* We may explore our periodic task handling, but at present this does not feel like it would have much of an operational gain for users of Ironic.
+<br>
+If your hoping a list including the node conductor can see a performance improvement, I wouldn't expect that unless we consider additional data model changes. But if you have a use case, please bring it to our attention and maybe it will inspire us!
```
Comment (jayofdoom) [inline src/pages/blog/performance-update.md:55]: node conductor? That doesn't make sense.

File: src/pages/blog/performance-update.md [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
+This whole effort challenged how we approached operations with node lists.
+<br>
+* Additional database indexes
+* Improvements to how we sanitize nodes
+* Improvements to how we handle the access control context when lists of nodes are being handled.
+* Removal of excess work from the Database, and Database client
+<br>
+
+![Graph of database and api nodes returned per second via the Ironic API"](/img/blog-performance-update.png)
```
Comment (jayofdoom) [inline src/pages/blog/performance-update.md:36]: This graph, in context of this blog post, only makes sense IMO if we demonstrate the old numbers with the new.

Comment (juliakreger) [reply src/pages/blog/performance-update.md:36]: That would be ironic 17.0 in the graph.

File: src/pages/blog/performance-update.md (hunk 2 above)
Comment (juliakreger) [reply src/pages/blog/performance-update.md:55]: Revising.

File: src/pages/blog/performance-update.md (hunk 3 above)
Comment (juliakreger) [reply src/pages/blog/performance-update.md:36]: Explicitly stressing 17.0 was just before we started this work.

File: src/pages/blog/performance-update.md [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+author: Julia Kreger
+date: 2021-07-09T22:00:00.000Z
+category:
+  - label: Development Update
+    id: category-A7fnZYrE9
+---
+
+## What a Journey ##
+
+When I started out on a journey of trying to improve the performance of ironic a few months ago, the journey ended up taking a bit of a different path from what I expected. But I can now say with confidence, becasue all of the patches have merged and even backported to our "Wallaby" release, that we've made Ironic faster.
```
Comment (elfosardo) [inline src/pages/blog/performance-update.md:13]: s/becasue/because

File: src/pages/blog/performance-update.md [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+* Removal of excess work from the Database, and Database client
+<br>
+These changes show an increadible performance improvement over Ironic 17.0 which was released before we started this effort to improve performance.
+<br>
+
+![Graph of database and api nodes returned per second via the Ironic API"](/img/blog-performance-update.png)
+
+### One size, does not fit everyone ###
+
+Ironic fulfills the needs of several different use cases. From a Systems Adminsitrator just needing to deploy the same OS image to a number of machines in fairly rapid order. To software deploying specialized ramdisk iamges and configuration such as those that may be used to stand up a small private cloud. To the largest and most complex deployments where Ironic powers the management of all Bare Metal nodes and resources within an organization.
```
Comment (elfosardo) [inline src/pages/blog/performance-update.md:42]: s/Adminsitrator/Administrator
s/iamges/images

**Review by elfosardo — CHANGES_REQUESTED** [review]

File: src/pages/blog/performance-update.md (hunk 4 above)
Comment (juliakreger) [reply src/pages/blog/performance-update.md:13]: Fixed

File: src/pages/blog/performance-update.md (hunk 5 above)
Comment (juliakreger) [reply src/pages/blog/performance-update.md:42]: fixed.

**Review by elfosardo — APPROVED** [review]

**Pull request merged**